rln/utils.go

```diff
@@ -98,3 +125,47 @@ func Bytes32ToBigInt(value [32]byte) *big.Int {
 	result.SetBytes(b)
 	return result
 }
+
+// Keccak functions take from here. To avoid unnecessary dependency to go-ethereum.
+// https://github.com/ethereum/go-ethereum/blob/v1.13.11/crypto/crypto.go#L62-L84
+
+// KeccakState wraps sha3.state. In addition to the usual hash methods, it also supports
+// Read to get a variable amount of data from the hash state. Read is faster than Sum
+// because it doesn't copy the internal state, but also modifies the internal state.
+type KeccakState interface {
+	hash.Hash
+	Read([]byte) (int, error)
+}
+
+// NewKeccakState creates a new KeccakState
+func NewKeccakState() KeccakState {
+	return sha3.NewLegacyKeccak256().(KeccakState)
+}
+
+// Keccak256 calculates and returns the Keccak256 hash of the input data.
+func Keccak256(data ...[]byte) []byte {
+	b := make([]byte, 32)
+	d := NewKeccakState()
+	for _, b := range data {
+		d.Write(b)
+	}
+	d.Read(b)
+	return b
+}
```

Comment:
Depending on how often will this hashing happens, it might be a good idea to use a `sync.Pool` similar to https://github.com/waku-org/go-waku/blob/master/waku/v2/hash/hash.go to reduce memory allocations

Reply:
right! great idea, fixed here